### PR TITLE
Error on schema with attribute names duplicated in types

### PR DESCRIPTION
**Pull request by john-h-kastner-aws**
This was the original behavior, but it regressed when schema type parsing was rewritten in #132. The regression was not noticed due to a faulty test case. This commit also fixes the test case, and adds new test cases for other places where a duplicate key might go unnoticed.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by aaronjeline — CHANGES_REQUESTED** [review]
Could you put a note in the CHANGELOG?

File: cedar-policy-validator/src/schema_file_format.rs [hunk 1]
```diff
@@ -474,10 +482,8 @@ pub enum SchemaTypeVariant {
         element: Box<SchemaType>,
     },
     Record {
-        #[serde(with = "serde_with::rust::maps_duplicate_key_is_error")]
         attributes: BTreeMap<SmolStr, TypeOfAttribute>,
```
Comment (cdisselkoen) [inline cedar-policy-validator/src/schema_file_format.rs:485]: does this need to be `AttributesTypeMap` instead of the `BTreeMap`?

File: cedar-policy-validator/src/schema_file_format.rs [hunk 2]
```diff
@@ -474,10 +482,8 @@ pub enum SchemaTypeVariant {
...
         #[serde(rename = "additionalAttributes")]
-        #[serde(default = "additional_attributes_default")]
```
Comment (cdisselkoen) [inline cedar-policy-validator/src/schema_file_format.rs:480]: why remove this?  and if we're removing this, should we also remove the function it refers to?

**Review by cdisselkoen — APPROVED** [review]
Tests look good

File: cedar-policy-validator/src/schema_file_format.rs (hunk 2 above)
Comment (john-h-kastner-aws) [reply cedar-policy-validator/src/schema_file_format.rs:480]: It's still called by the manual deserialization implementation. It's removed because the annotation is dead code since were not deriving deserialize.

File: cedar-policy-validator/src/schema_file_format.rs (hunk 1 above)
Comment (john-h-kastner-aws) [reply cedar-policy-validator/src/schema_file_format.rs:485]: It could, but I think I prefer leaving that struct as just an implementation detail for the custom deserializer. I'd rather not have the struct at all, so this makes it possible to delete it later if I figure out has serde_with works without messing with any other code.
